src/main/java/no/uib/inf112/core/player/PlayerHandler.java

```diff
@@ -50,76 +50,30 @@ public PlayerHandler(int playerCount, MapHandler map) {
         for (IPlayer player : players) {
             player.setDock(docks.pop());
         }
-
-    }
-
-    @Override
-    public void generatePlayers() {
-        //Keeping this in case we want to generate new players, currently only used for testing
-        if (HEADLESS) {
-            players = new ArrayList<>(playerCount);
-            for (int i = 0; i < playerCount; i++) {
-                players.add(new NonPlayer(5 + i, 2, Direction.NORTH, map));
-            }
-
-            Stack<Integer> docks = new Stack<>();
-            for (int i = 1; i <= playerCount; i++) {
-                docks.push(i);
-            }
-            Collections.shuffle(docks);
-
-            for (IPlayer player : players) {
-                player.setDock(docks.pop());
-            }
-        }
-
-    }
-
-    @Override
-    public void generateOnePlayer() {
-        players = new ArrayList<>(playerCount);
-        players.add(new NonPlayer(0, 0, Direction.NORTH, map));
-        players.get(0).setDock(1);
     }
 
     @Override
     public void endTurn() {
-        for (int i = 0; i < Player.MAX_PLAYER_CARDS; i++) {
-            List<PlayerCard> cards = new ArrayList<>();
-            for (IPlayer p : players) {
-                cards.add(p.getNextCard(i));
-            }
-            Collections.sort(cards);
-            for (int j = 0; j < cards.size(); j++) {
-                PlayerCard card = cards.get(j);
-
-                GameGraphics.executorService.schedule(() ->
-                        Gdx.app.postRunnable(() ->
-                                card.getPlayer().moveRobot(card.getCard().getAction())), 500 * (i + 1), TimeUnit.MILLISECONDS);
-            }
-        }
-
-        GameGraphics.executorService.schedule(() ->
-                Gdx.app.postRunnable(() -> GameGraphics.getRoboRally().round()), 500 * (Player.MAX_PLAYER_CARDS + 2), TimeUnit.MILLISECONDS);
+        GameGraphics.getRoboRally().round();
     }
 
     @Override
     public void startTurn() {
-        //TODO Issue #44 check if dead
-        //TODO Issue #44 check if player is out side of map
 
-        GameGraphics.getRoboRally().getDeck().shuffle();
-        UserPlayer p = mainPlayer();
-        if (p.isPoweredDown()) {
-            //TODO Issue #24 check if is powered down (then heal)
+        Player p = (Player) mainPlayer();
+        if (p.isDestroyed()) {
             return;
+        }
+        if (p.isPoweredDown()) {
+            p.heal();
```

Comment:
Heal "MAX_HEALTH" not Integer.MAXVALUE. Player should not have that kind of health

Reply:
It should be capped by heal(integer), this way we can have entities with different amount of max health and they will always be healed fully